calico_kubernetes/policy.py

```diff
@@ -98,34 +98,24 @@ def _generate_rule(self, arguments):
         if protocol:
             rule_args["protocol"] = protocol
         if src_ports:
-            rule_args["src_ports"] = self._validate_ports(src_ports)
+            rule_args["src_ports"] = src_ports
```

Comment:
I think we still need something here to convert to a list, correct?

`src_ports` is a string which looks like this: `"10,20,30"` but it needs to looks like this: `[10, 20, 30]`